src/python/grpcio/grpc/_cython/_cygrpc/channel.pyx.pxi

```diff
@@ -233,18 +234,31 @@ cdef void _call(
   cdef grpc_call_error c_call_error
   cdef tuple error_and_wrapper_tag
   cdef _BatchOperationTag wrapper_tag
+  cdef void *registered_call_handle
+  cdef const char *host_ptr
   with channel_state.condition:
     if channel_state.open:
       method_slice = _slice_from_bytes(method)
       if host is None:
         host_slice_ptr = NULL
+        host_ptr = NULL
       else:
         host_slice = _slice_from_bytes(host)
         host_slice_ptr = &host_slice
-      call_state.c_call = grpc_channel_create_call(
-          channel_state.c_channel, NULL, flags,
-          c_completion_queue, method_slice, host_slice_ptr,
-          _timespec_from_time(deadline), NULL)
+        host_ptr = <const char *>host
+      if registered_method:
+        registered_call_handle = grpc_channel_register_call(
+          channel_state.c_channel, <const char *>method,
+          host_ptr, NULL)
+        call_state.c_call = grpc_channel_create_registered_call(
+            channel_state.c_channel, NULL, flags,
+            c_completion_queue, registered_call_handle,
+            _timespec_from_time(deadline), NULL)
+      else:
+        call_state.c_call = grpc_channel_create_call(
```

Comment:
I'm worried that with this change, the vast majority of our testing will go down this branch, but the vast majority of our users will be going down the other branch. Can we either parameterize our existing tests on method registration or just manually set `registered` in most of them?